### PR TITLE
Remove release workflow

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

We consider lading's container to be the only supported release artifact at this time. If a use-case for binaries or the crates.io artifacts appear in the future we'll be glad to re-introduce them.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,103 +0,0 @@
...
-  create-release:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-
-      - name: Create GitHub release
-        uses: taiki-e/create-gh-release-action@72d65cee1f8033ef0c8b5d79eaf0c45c7c578ce3 # v1.8.2
-        with:
-          changelog: CHANGELOG.md
-          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (GeorgeHahn) [inline .github/workflows/release.yml:21]: Thoughts on retaining this piece? I find the changelogs helpful at least.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,103 +0,0 @@
...
-
-  upload-binaries:
```
Comment (GeorgeHahn) [inline .github/workflows/release.yml:23]: Could you also remove the readme note about binaries? https://github.com/DataDog/lading?tab=readme-ov-file#binary-releases

Comment (GeorgeHahn) [reply .github/workflows/release.yml:23]: Looking at this again, it'd be helpful to have a link to lading on GHCR: https://github.com/DataDog/lading/pkgs/container/lading.

File: .github/workflows/release.yml (hunk 1 above)
Comment (blt) [reply .github/workflows/release.yml:21]: I'll put this back.

Comment (blt) [reply .github/workflows/release.yml]: Resolved in [bb68f5e](https://github.com/DataDog/lading/pull/1089/commits/bb68f5e08a442a68087aafd9379cc7a966eb7c50)

**Review by GeorgeHahn — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (GeorgeHahn) [reply .github/workflows/release.yml:23]: Ideal!

**Pull request merged**